docs/dg/dev/architecture/coding-conventions.md

```diff
@@ -0,0 +1,2165 @@
+        │   │   └── [Module]To[Module]ServiceInterface.php           
+        │   └── Plugin
+        │       └── [PluginInterfaceName]PluginInterface.php
+        ├── Plugin
+        │   ├── Provider
+        │   │   ├── [Name]ControllerProvider.php
+        │   │   └── [Name]ServiceProvider.php
+        │   └── [ConsumerModule]
+        │       └── [PluginName]Plugin.php
+        ├── [Module]Config.php
+        ├── [Module]DependencyProvider.php
+        └── [Module]Factory.php
+```
+
+Used components
+- [Bridge](#bridge)
+- [Config](#module-configurations)
+- [Controller](#controller)
+- [Dependency Provider](#dependency-provider)
+- [Factory](#factory)
+- [Mapper / Expander / Hydrator](#mapper--expander--hydrator)
+- [Model](#model)
+- [Plugin](#plugin), [Plugin Interface](#plugin-interface)
+
+## Client
+
+`Client` is a lightweight application layer that handles all data access, such as
+- Persistence access (such as Key-Value Storage (Redis), Search (Elasticsearch)),
+- Yves sessions,
+- `Zed` as a data-source (RPC),
+- 3rd party communication.
+
+Note: Backend database access is an exception for performance streamlining.
+
+```
+[Organization]
+└── Client
+    └── [Module]
+        ├── [CustomDirectory]
+        │   ├── [ModelName]Interface.php
+        │   └── [ModelName].php        
+        ├── Dependency
+        │   ├── Client
+        │   │   ├── [Module]To[Module]ClientBridge.php
+        │   │   └── [Module]To[Module]ClientInterface.php
+        │   ├── Service
+        │   │   ├── [Module]To[Module]ServiceBridge.php
+        │   │   └── [Module]To[Module]ServiceInterface.php           
+        │   └── Plugin
+        │       └── [PluginInterfaceName]PluginInterface.php            
+        ├── Plugin
+        │   └── [ConsumerModule]
+        │       └── [PluginName]Plugin.php
+        ├── Zed
+        │   ├── [Module]Stub.php
+        │   └── [Module]StubInterface.php
+        ├── [CustomDirectory]
+        ├── [Module]ClientInterface.php
+        ├── [Module]Client.php
+        ├── [Module]Config.php
+        ├── [Module]DependencyProvider.php
+        └── [Module]Factory.php
+```
+
+Used components
+- [Bridge](#bridge)
+- [Client facade](#client-facade)
+- [Config](#module-configurations)
+- [Dependency Provider](#dependency-provider)
+- [Factory](#factory)
+- [Mapper / Expander / Hydrator](#mapper--expander--hydrator)
+- [Model](#model)
+- [Plugin](#plugin), [Plugin Interface](#plugin-interface)
+- [Zed Stub](#zed-stub)
+
+## Service
+
+The `Service` application layer encapsulates a multipurpose library that can be utilized across various application layers, such as `Yves` or `Zed` application layers.
+A `Service` primarily consists of reusable lightweight stateless business logic components.
+Due to its deployment across all applications, a `Service` is constrained to accessing data providers that are available universally (for instance, the backend database is not accessible from storefront applications by default).
```

Comment:
we do, but resources are not wired to all applications.

If you write a `DBService::saveDBData($table, $row)`, then you can call this service from Yves & Zed, because the Service code is deployed in all applications, but Yves will throw you an exception when it can't reach the relational database.